### PR TITLE
feat: show build number value in the About if present in the config

### PR DESCRIPTION
**Pull request by cccs-joel**
### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
aboutafter

### TESTING INSTRUCTIONS
Add a variable called BUILD_NUMBER in the config.py, the value should be displayed in the About section underneath the version/sha. No build information is shown if there is no variable BUILD_NUMBER in the config.py file.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [X] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by cccs-joel** [comment]
Hi, do I need to action something from my side to speed things up?

**Comment by cccs-rc** [comment]
Hi there! Is there anything we can do to help get this merged? @junlincc? @nytai?

**Review by dparent1 — APPROVED** [review]

**Comment by dparent1** [comment]
My mistake for reviewing, apologies.

**Comment by cccs-joel** [comment]
How can we get the ball rolling with this PR? @junlincc

**Comment by nytai** [comment]
Could you please explain your use case and why this is needed? Does the current version config, kept in package.json, not meet your needs?

**Comment by cccs-joel** [comment]
Sure, we want to qualify/augment the version number available in the config by adding a build number shown in the about. Would be useful for our QA people. That number in our case is the date we build the image and a sequence number, too long to fit on the same line of the Superset version.

**Comment by nytai** [comment]
Ok, looks like CI is failing. Can you add a default value to config.py and a comment explaining what this config does and how to use it?

File: superset/views/base.py [hunk 1]
```diff
@@ -312,6 +312,11 @@ def menu_data() -> Dict[str, Any]:
     brand_text = appbuilder.app.config["LOGO_RIGHT_TEXT"]
     if callable(brand_text):
         brand_text = brand_text()
+    build_number = ""
+    try:
+        build_number = appbuilder.app.config["BUILD_NUMBER"]
+    except Exception as ex:
+        logger.debug("BUILD_NUMBER is missing from the config", ex)
```
Comment (nytai) [inline superset/views/base.py:319]: this shouldn't be necessary if you add a default value in config.py, (either empty string or `None`). Config values should never be undefined.

Comment (cccs-joel) [reply superset/views/base.py]: Done.

Comment (cccs-joel) [reply superset/views/base.py]: Anything required at our end at this point?

Comment (nytai) [reply superset/views/base.py]: Yes, you need to get CI passing before we can merge

**Review by nytai — APPROVED** [review]

**Comment by cccs-joel** [comment]
There is still a failing check, no clue why. All I get is: T_he operation was canceled._ for Python Misc / python-lint

**Comment by nytai** [comment]
Yes. I think there’s an issue with a pylint upgrade that’s currently affecting all PRs.

**Comment by nytai** [comment]
@cccs-joel you might have to rebase your branch to fix, seems like some of the more recently opened PRs are not having issues.

**Comment by cccs-joel** [comment]
@nytai I just merged master into my branch once again, hopefully that will trigger the checks and they will all pass.

**Comment by cccs-joel** [comment]
@nytai Ok, that did the trick, all checks are now passing. Can someone merge?

**Pull request merged**